.github/workflows/apidocs-upload.yml

```diff
@@ -0,0 +1,85 @@
+name: apidocs
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+
+  # First check if this version is a valid public release
+  check-version:
+    runs-on: ubuntu-20.04
+    outputs:
+      release_status_output: ${{ steps.release_version.outputs.version_status_output }}
+
+    steps:
+      - name: Check release version
+        id: release_version
+        shell: bash
+        run: |
+  
+          if [[ "${{ github.ref }}" = refs/tags/* ]]; then
+            version="${ref#refs/tags/}"
+            # This patterns accepts "v1.0.0" , "v2.4.6". Rejects "v1.0.0-rc1".
+            if [[ "$version" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+              result=isRelease
+            else
+              result=isNotRelease
+            fi
+          else
+            result=isNotTag
+          fi
+  
+          echo $result
+          echo "version_status_output=$result" >> $GITHUB_OUTPUT
+
+  # Initiate upload only if the version is valid
+  upload-apidocs:
+    runs-on: ubuntu-20.04
+    needs: check-version
+    if: ${{ needs.test_version.outputs.release_status_output == 'isRelease' }}
```

Comment:
We should consider that we will have a `dev` environment as well. We need to deploy both the `dev` and `prod`, according to what branch/tag trigger the build.

I would say we should deploy to `dev` when there is a commit in the `master` branch, and to `prod` using your current logic.

Reply:
I'm under the impression that uploading at every commit to `master` would be confusing to our current deployment structure.

On [6c051a7](https://github.com/HathorNetwork/hathor-wallet-headless/pull/364/commits/6c051a71aaf053e03754eadf005e35c9425f7cc4) I implemented a system where the _release candidates_ are sent to a staging environment. What do you think about this approach?

Reply:
I think it's good